src/parcsr_ls/par_nodal_systems.c

```diff
@@ -798,74 +798,64 @@ hypre_BoomerAMGCreateNodalA(hypre_ParCSRMatrix    *A,
+   hypre_CSRMatrix    *A_diag = hypre_ParCSRMatrixDiag(A);
+   HYPRE_Int          *A_diag_i = hypre_CSRMatrixI(A_diag);
+   HYPRE_Int          *A_diag_j = hypre_CSRMatrixJ(A_diag);
+   HYPRE_Complex      *A_diag_data = hypre_CSRMatrixData(A_diag);
+   hypre_CSRMatrix    *A_offd = hypre_ParCSRMatrixOffd(A);
+   HYPRE_Int          *A_offd_i = hypre_CSRMatrixI(A_offd);
+   HYPRE_Int          *A_offd_j = hypre_CSRMatrixJ(A_offd);
+   HYPRE_Complex      *A_offd_data = hypre_CSRMatrixData(A_offd);
    hypre_CSRMatrix    *SN_diag = hypre_ParCSRMatrixDiag(SN);
    HYPRE_Int          *SN_diag_i = hypre_CSRMatrixI(SN_diag);
    HYPRE_Int          *SN_diag_j = hypre_CSRMatrixJ(SN_diag);
-   HYPRE_Real         *SN_diag_data;
    hypre_CSRMatrix    *SN_offd = hypre_ParCSRMatrixOffd(SN);
    HYPRE_Int          *SN_offd_i = hypre_CSRMatrixI(SN_offd);
    HYPRE_Int          *SN_offd_j = hypre_CSRMatrixJ(SN_offd);
-   HYPRE_Real         *SN_offd_data = NULL;
    HYPRE_Int          *CF_marker;
    HYPRE_BigInt       *col_map_offd_SN = hypre_ParCSRMatrixColMapOffd(SN);
-   HYPRE_BigInt       *col_map_offd_S;
+   HYPRE_BigInt       *col_map_offd_A = hypre_ParCSRMatrixColMapOffd(A);
+   HYPRE_BigInt       *col_map_offd_S = NULL;
    HYPRE_Int          *dof_func;
    HYPRE_Int           num_nodes = hypre_CSRMatrixNumRows(SN_diag);
    HYPRE_Int           num_variables;
-   hypre_ParCSRCommPkg *comm_pkg = hypre_ParCSRMatrixCommPkg(SN);
-   HYPRE_Int           num_sends;
-   HYPRE_Int           num_recvs;
-   HYPRE_Int          *send_procs;
-   HYPRE_Int          *send_map_starts;
-   HYPRE_Int          *send_map_elmts;
-   HYPRE_Int          *recv_procs;
-   HYPRE_Int          *recv_vec_starts;
-   hypre_ParCSRCommPkg *comm_pkg_S;
-   HYPRE_Int          *send_procs_S;
-   HYPRE_Int          *send_map_starts_S;
-   HYPRE_Int          *send_map_elmts_S;
-   HYPRE_Int          *recv_procs_S;
-   HYPRE_Int          *recv_vec_starts_S;
-   HYPRE_Int          *col_offd_S_to_A = NULL;
+   HYPRE_Int          *S_marker;
+   HYPRE_Int          *S_marker_offd = NULL;
+   HYPRE_Int          *S_tmp_j;
 
    HYPRE_Int           num_coarse_nodes;
-   HYPRE_Int           i,j,k,k1,jj,cnt;
-   HYPRE_BigInt        big_k1;
-   HYPRE_Int           row, start, end;
+   HYPRE_Int           i,j,k,cnt;
    HYPRE_Int           num_procs;
-   HYPRE_Int           num_cols_offd_SN = hypre_CSRMatrixNumCols(SN_offd);
-   HYPRE_Int           num_cols_offd_S;
-   HYPRE_Int           SN_num_nonzeros_diag;
-   HYPRE_Int           SN_num_nonzeros_offd;
+   HYPRE_Int           num_cols_offd_A = hypre_CSRMatrixNumCols(A_offd);
+   HYPRE_Int           A_num_nonzeros_diag;
+   HYPRE_Int           A_num_nonzeros_offd;
    HYPRE_Int           S_num_nonzeros_diag;
    HYPRE_Int           S_num_nonzeros_offd;
    HYPRE_BigInt        global_num_vars;
    HYPRE_BigInt        global_num_cols;
    HYPRE_BigInt        global_num_nodes;
+   HYPRE_Int           nnz, S_cnt, in;
```

Comment:
This is not a new issue in this PR. Line 871 `if (nodal < 0) {}`: 1. not sure what's the meaning of `nodal < 0`, probably not documented. 2. `dof_func` is allocated as size of `num_coarse_nodes*num_functions` only for C points, which does not match the comment above this function, "scalar version of dof_array"